src/swap/useSwapQuote.ts

```diff
@@ -204,22 +202,20 @@ function useSwapQuote(networkId: NetworkId, slippagePercentage: string) {
       return quoteResult
     },
     {
-      onSuccess: (updatedExchangeRate: QuoteResult | null) => {
-        setExchangeRate(updatedExchangeRate)
-      },
+      // Keep last result when refreshing
+      setLoading: (state) => ({ ...state, loading: true }),
       onError: (error: Error) => {
-        setExchangeRate(null)
```

Comment:
in the event that a user has a quote -> updates the swap pair -> the quote fetch produces an error, in this old flow the quote would be cleared but now we would continue to show the previous quote - do I read this correctly / is this intentional?

Reply:
The quote is cleared if there's an error.
It's only kept while loading, not if there's an error.

Reply:
oh i see! thanks!